portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-validator.js

```diff
@@ -95,46 +96,155 @@ const validateCsvData = (csvData) => {
   return validationErrors;
 };
 
+/**
+ * Gets the
+ * @param {import('@ukef/dtfs2-common').OneIndexedMonth} oneIndexedMonth - The one indexed month
+ * @returns {string} A regex string for the given month and year combination
+ * @throws {Error} If a month matching the supplied `longMonthName` cannot be found
+ */
+const getMonthRegexString = (oneIndexedMonth) => {
+  const monthNames = Object.values(MONTH_NAMES).find(({ index }) => index === oneIndexedMonth);
+  if (!monthNames) {
+    throw new Error(`Failed to find month names for one indexed month '${oneIndexedMonth}'`);
+  }
+  return `(${monthNames.long}|${monthNames.short}|${monthNames.numeric})`;
+};
+
+/**
+ * Gets the regex instance for the supplied groups using the
+ * format which is expected for the filename
+ * @param  {...string} groups - The month and year components
+ * @returns {RegExp} The regex for the supplied months and years
+ * @example
+ * // Monthly, January 2024
+ * const monthRegexString = `(${'January'}|${'Jan'}|${'01'})`;
+ * const year = 2024;
+ * const filenameReportPeriodRegex = getFilenameReportPeriodRegex(
+ *   monthRegexString,
+ *   year,
+ * ); // equivalent to '/(January|Jan|01)[-_]2024/i'
+ *
+ * // Quarterly, November 2023 to February 2024
+ * const startMonthRegexString = `(${'November'}|${'Nov'}|${'11'})`;
+ * const startYear = 2023;
+ * const endMonthRegexString = `(${'February'}|${'Feb'}|${'02'})`;
+ * const endYear = 2024;
+ * const filenameReportPeriodRegex = getFilenameReportPeriodRegex(
+ *   startMonthRegexString,
+ *   startYear,
+ *   endMonthRegexString,
+ *   endYear,
+ * ); // equivalent to '/(November|Nov|11)[-_]2023[-_](February|Feb|02)[-_]2024/i'
+ */
+const getFilenameReportPeriodRegex = (...groups) => {
+  const regexString = groups.join(FILE_UPLOAD.FILENAME_SEPARATOR_GROUP);
+  return new RegExp(regexString, 'i');
+};
+
+/**
+ * Gets an example filename for the specified report period
+ * @param {import('@ukef/dtfs2-common').ReportPeriod} dueReportPeriod - The due report period
+ * @returns {string} An example filename
+ */
+const getExampleFilenameReportPeriod = (dueReportPeriod) => {
+  let exampleFilename;
+  if (isEqualMonthAndYear(dueReportPeriod.start, dueReportPeriod.end)) {
+    exampleFilename = `${dueReportPeriod.start.month}-${dueReportPeriod.start.year}`;
+  } else if (dueReportPeriod.start.year === dueReportPeriod.end.year) {
+    exampleFilename = `${dueReportPeriod.start.month}-${dueReportPeriod.end.month}-${dueReportPeriod.start.year}`;
+  } else {
+    exampleFilename = `${dueReportPeriod.start.month}-${dueReportPeriod.start.year}-${dueReportPeriod.end.month}-${dueReportPeriod.end.year}`;
+  }
+
+  const singleDigitMonthRegex = /(?<!\d)(\d{1}-)/g;
+  const replacePattern = '0$1';
+  return exampleFilename.replaceAll(singleDigitMonthRegex, replacePattern);
```

Comment:
This code will match any single digit followed by a `'-'` character which is not preceded by a digit, and then append a `'0'` character to the matched group in the string